sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -98,14 +98,15 @@ static az_result root(az_event_policy* me, az_event event)
 /**
  * @brief start subscription timer
  */
-AZ_INLINE az_result _rpc_start_timer(az_mqtt5_rpc_server* me)
+AZ_INLINE az_result _rpc_start_timer(az_mqtt5_rpc_server_policy* me)
 {
   _az_event_pipeline* pipeline = &me->_internal.connection->_internal.event_pipeline;
   _az_event_pipeline_timer* timer = &me->_internal.rpc_server_timer;
 
   _az_RETURN_IF_FAILED(_az_event_pipeline_timer_create(pipeline, timer));
 
-  int32_t delay_milliseconds = (int32_t)me->_internal.options.subscribe_timeout_in_seconds * 1000;
+  int32_t delay_milliseconds
```

Comment:
why not use size_t?

Reply:
Needs to be `int32_t` for `az_platform_timer_start`